samples/multiple-inheritance/sample.md

```diff
@@ -0,0 +1,20 @@
+## Introduction to Consumer Driven Contracts
+This sample shows a [consumer driven contracts](http://martinfowler.com/articles/consumerDrivenContracts.html)(CDD) approach to messaging. The essense of consumer driven contracts is that the owner ship of the contract is inversed. Instead of the producer defining the contract consumers are now the ones defining the contract they expect and it's up to the producer to fullfill it.
+
+In the NServiceBus world message contracts is plain C# types so to honor a consumer contract the producer would just make the relevant message contract inherit from the contract type. 
+
+### Contracts as interfaces
+
+CDD is the main reason that `interface` messages are supported in addition to classes and the reason is that more than one consumer can provide a contract that would be satisfied by the same publisher message. If only classes where used this would not be possible since multiple inheritance isn't supported by C#. The solution is to use interfaces instead since with interfaces the publisher could implement all relevant contract types on the same message type.
+
+Assuming that the producers `MyEvent` would satisfy both Consumer1Contract and Consumer2Contract this would look like this:
```

Comment:
Can we use different event names, instead of `MyEvent`. It would be great if we can name the events after a well known event, eg. in the retail vertical?